### ISSUE TITLE
Cake's logic issue

### ISSUE THREAD

**Issue by caoli5288** [PocketMine/PocketMine-MP#3753]
Player's EntityRegainHealthEvent can be cancelled by plugin, but the cake still consumed.
```php
public function onActivate(Item $item, Player $player = null){
    if($player instanceof Player and $player->getHealth() < $player->getMaxHealth()){
        ++$this->meta;
	$ev = new EntityRegainHealthEvent($player, 3, EntityRegainHealthEvent::CAUSE_EATING);
	$player->heal($ev->getAmount(), $ev);

	if($this->meta >= 0x06){
		$this->getLevel()->setBlock($this, new Air(), true);
	}else{
		$this->getLevel()->setBlock($this, $this, true);
	}
	return true;
    }
    return false;
}
```
Fix please.

**Comment by Creeperface01** [comment]
ok

**Comment by PEMapModder** [comment]
ref: #3756

**Comment by PEMapModder** [comment]
Resolved in #3783
